Add "Coursework PR Guide" for trainee

cjyuan opened the pull request:
## What does this change?

Added a PR guide for trainees.

### Org Content?

Added `org-cyf-guides/content/trainee-pr-guide`

## Checklist

- [x] I have read the [contributing guidelines](CONTRIBUTING.MD)
- [x] I have checked my spelling and grammar with an [automated tool](https://www.grammarly.com/grammar-check)
- [x] I have previewed my changes to check the [markdown renders](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) as I intend

## Who needs to know about this?
@illicitonion

Commit:
Add Trainee PR Guides

cjyuan committed:
Merge branch 'CodeYourFuture:main' into feature/pr-guide

cjyuan committed:
Resolve unresolved merge conflict

illicitonion commented on org-cyf-guides/content/trainee-pr-guide/_index.md:
```diff
@@ -0,0 +1,26 @@
+A **pull request**, or **PR**, is a way to share your code for review.
+In professional projects, PRs are used to suggest changes and then merge them into the main project.
+
+At CodeYourFuture, we use PRs mainly as a way for you to submit your work for review and feedback.
```
```suggestion
At {{<our-name>}}, we use PRs mainly as a way for you to submit your work for review and feedback.
```

illicitonion commented on org-cyf-guides/content/trainee-pr-guide/opening-a-pr/index.md:
```diff
@@ -0,0 +1,87 @@
++++
+title = "Opening PR"
```
```suggestion
title = "Opening a PR"
```

illicitonion commented on org-cyf-guides/content/trainee-pr-guide/opening-a-pr/index.md:
```diff
@@ -0,0 +1,87 @@
+weight = 2
++++
+
+### Selecting source and target branches
```
This feels like it could do with a screenshot?

cjyuan replied:
Added a screenshot.

illicitonion commented on org-cyf-guides/content/trainee-pr-guide/opening-a-pr/index.md:
```diff
@@ -0,0 +1,87 @@
++++
+
+### Selecting source and target branches
+- **Source branch**: Your branch
```
We probably want to mention that they should make sure they've selected their fork?

cjyuan replied:
Because of the way they open a PR, GitHub would normally select the correct source repo.

The normal mistakes I came across were:
- They selected their own `main` as the target branch
- Instead of `main`, GitHub selected a branch in CYF repo that matches the trainee's branch name (Few branches in the Module-Data-Flow were created by trainees)
-

illicitonion commented on org-cyf-guides/content/trainee-pr-guide/opening-a-pr/index.md:
```diff
@@ -0,0 +1,87 @@
+means the title should have 5 components separated by a vertical bar character (`|`)
+
+- **`Region`**: One of 
+  - `Cape Town`, `South Africa`, `ZA`
+  - `Glasgow`, `Scotland`
+  - `London`
+  - `North West`, `NW`, `Manchester`
+  - `Sheffield`
+  - `West Midlands`, `WM`, `Birminghams`
```
```suggestion
  - `Cape Town`
  - `Glasgow`
  - `London`
  - `Manchester`
  - `Sheffield`
  - `Birmingham`
```

illicitonion commented on org-cyf-guides/content/trainee-pr-guide/opening-a-pr/index.md:
```diff
@@ -0,0 +1,87 @@
+- **`Sprint`**: `Sprint 1`, `Sprint 2`, ...
+- **`Assignment Title`**:
+  - Project's name
+  - "Coursework" for generic coursework exercise
```
```suggestion
  - The name of the backlog issue is generally a good thing to use.
```

illicitonion commented on org-cyf-guides/content/trainee-pr-guide/using-labels/index.md:
```diff
@@ -0,0 +1,30 @@
+  - Replaces `Needs review` and `Review in progress`
+
+- `Complete`
+  - Added by a **mentor** when the PR has been approved
```
```suggestion
  - Added by a **mentor** when the PR has been approved and no more work is required on this PR
```

illicitonion commented on org-cyf-guides/content/trainee-pr-guide/using-labels/index.md:
```diff
@@ -0,0 +1,30 @@
+
+- `Reviewed`
+  - Added by a **mentor** once the PR has been reviewed and the trainee can start working 
+    on feedback
```
```suggestion
    on feedback - this indicates there is something more for the trainee to do.
```

illicitonion reviewed (commented):
This looks great, and really useful, thanks!

I'd suggest maybe re-organising these guides a little - we have a "Reviewing" guide already - maybe we could rename reviewing to "Code Review", and have the "Trainee PR Guide" be a child directory of the "Code Review" guide?

cjyuan committed:
Update org-cyf-guides/content/trainee-pr-guide/using-labels/index.md

Co-authored-by: Daniel Wagner-Hall <dawagner@gmail.com>

cjyuan committed:
Update org-cyf-guides/content/trainee-pr-guide/using-labels/index.md

Co-authored-by: Daniel Wagner-Hall <dawagner@gmail.com>

cjyuan committed:
Update org-cyf-guides/content/trainee-pr-guide/opening-a-pr/index.md

Co-authored-by: Daniel Wagner-Hall <dawagner@gmail.com>

cjyuan committed:
Update org-cyf-guides/content/trainee-pr-guide/opening-a-pr/index.md

Co-authored-by: Daniel Wagner-Hall <dawagner@gmail.com>

cjyuan committed:
Update org-cyf-guides/content/trainee-pr-guide/opening-a-pr/index.md

Co-authored-by: Daniel Wagner-Hall <dawagner@gmail.com>

cjyuan committed:
Update org-cyf-guides/content/trainee-pr-guide/_index.md

Co-authored-by: Daniel Wagner-Hall <dawagner@gmail.com>

cjyuan committed:
Update org-cyf-guides/content/trainee-pr-guide/_index.md

Co-authored-by: Daniel Wagner-Hall <dawagner@gmail.com>

cjyuan committed:
Merge branch 'CodeYourFuture:main' into feature/pr-guide

Commit:
Make Trainee PR Guide under Reviewing Guide

- Rename "Reviewing" to "Code Review"
- Add a "Review Process" section to "Using Labels" guide
- Combine "Keeping Branch Clean", "Open a PR", and "Addressing Feedback"
  into one guide.

cjyuan commented:
I incorporated your suggested change, added a "Review process" section in "Using Labels" Guide.

For the Trainee PR Guide, I mainly combined the separate guides I prepared into one, and move them under the "Code Review" guide.

illicitonion reviewed (approved):
Perfect - thank you!
